homeassistant/components/ialarmxr/alarm_control_panel.py

```diff
@@ -0,0 +1,63 @@
+"""Interfaces with iAlarmXR control panels."""
+from __future__ import annotations
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IAlarmXRDataUpdateCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up a iAlarmXR alarm control panel based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([IAlarmXRPanel(coordinator)])
+
+
+class IAlarmXRPanel(CoordinatorEntity, AlarmControlPanelEntity):
+    """Representation of an iAlarmXR device."""
+
+    _attr_supported_features = (
+        AlarmControlPanelEntityFeature.ARM_HOME
+        | AlarmControlPanelEntityFeature.ARM_AWAY
+    )
+    _attr_name = "iAlarmXR"
+    _attr_icon = "mdi:security"
+
+    def __init__(self, coordinator: IAlarmXRDataUpdateCoordinator) -> None:
+        """Initialize the alarm panel."""
+        super().__init__(coordinator)
+        self.coordinator: IAlarmXRDataUpdateCoordinator = coordinator
+        self._attr_unique_id = coordinator.mac
+        self._attr_device_info = DeviceInfo(
+            manufacturer="Antifurto365 - Meian",
+            name=self.name,
+            connections={(device_registry.CONNECTION_NETWORK_MAC, coordinator.mac)},
+        )
+
+    @property
+    def state(self) -> str | None:
+        """Return the state of the device."""
+        return self.coordinator.state
+
+    def alarm_disarm(self, code: str | None = None) -> None:
```

Comment:
Please add this file to `.coveragerc` if you don't intend to add tests for it.

Reply:
Sorry @bdraco I wasn't able to improve more than 67% the code coverage in config flow test, but I think I have verified all the branches...I don't know why it's not 100% covered.

Reply:
This is a report I created today after the test improvement
<img width="742" alt="Screenshot 2022-05-18 at 12 43 39" src="https://user-images.githubusercontent.com/11453002/169021373-dc09da3c-147c-43cd-b3c4-088a32bdfb22.png">